src/post.ts

```diff
@@ -1,16 +1,35 @@
 import { setFailed } from '@actions/core'
-import { mkdirP, mv } from '@actions/io'
+import { mkdirP, mv, cp, rmRF } from '@actions/io'
 
-import { getVars } from './lib/getVars'
+import { Strategy, getVars } from './lib/getVars'
 import { isErrorLike } from './lib/isErrorLike'
 import log from './lib/log'
+import { exists } from '@actions/io/lib/io-util'
 
 async function post(): Promise<void> {
   try {
-    const { cacheDir, targetPath, cachePath } = getVars()
+    const { cacheDir, targetPath, cachePath, options } = getVars()
 
     await mkdirP(cacheDir)
-    await mv(targetPath, cachePath, { force: true })
+
+    switch (options.strategy) {
+      case Strategy.CopyImmutable:
+        if (await exists(cachePath)) {
+          log.info(`Cache already exists, skipping`)
+          return
+        }
+        await cp(targetPath, cachePath, { copySourceDirectory: true, recursive: true })
+        break
+      case Strategy.Copy:
+        await rmRF(cachePath)
+        await cp(targetPath, cachePath, { copySourceDirectory: true, recursive: true })
+        break
+      case Strategy.Move:
+        await mv(targetPath, cachePath, { force: true })
+        break
+    }
+
+    log.info(`Cache saved to ${cachePath} with ${options.strategy} strategy`)
```

Comment:
Nice touch adding some feedback about what's happening. We should make it consistent, tho.

In the last log it is saying CACHE SAVED even if it just said CACHE EXISTS SO WE SKIPPING.

I would suggest to add a log line inside each case. For the `copy-immutable` one you would log either SKIPPED or SAVED based on the condition you have there already.

Reply:
@stefanmaric on line 18, it returns immediately, so it should not print the CACHE SAVED line.
I can add a log line to each one however it will be a big redundant (3x same log?)

Reply:
Thank you @oNaiPs ! Need this change so much, makes a lot of sense for my use case. I hope @stefanmaric will have some time to approve this :)

Reply:
Yesterday I got error `EXDEV: cross-device link not permitted` when adding `action-local-cache` to my project. Many thank you to @oNaiPs, you fixes saved my life! 👍